packages/core-bridge/src/conversions.rs

```diff
@@ -182,52 +181,59 @@ impl ObjectHandleConversionsExt for Handle<'_, JsObject> {
-                    }
-                };
-            }
             if let Some(ref prom) = js_optional_getter!(cx, metrics, "prometheus", JsObject) {
-                                let addr = js_value_getter!(cx, prom, "bindAddress", JsString);
+                if js_optional_getter!(cx, metrics, "otel", JsObject).is_some() {
+                    cx.throw_type_error(
+                        "Invalid telemetryOptions.metrics: can't have both premetheus and otel at the same time",
+                    )?
+                }
+
+                let mut options = PrometheusExporterOptionsBuilder::default();
+
+                let addr = js_value_getter!(cx, prom, "bindAddress", JsString);
                 match addr.parse::<SocketAddr>() {
-                    Ok(address) => telemetry_opts.metrics(MetricsExporter::Prometheus(address)),
-                    Err(_) => cx.throw_type_error(
+                    Ok(addr) => options.socket_addr(addr),
+                    Err(_) => return cx.throw_type_error(
                         "Invalid telemetryOptions.metrics.prometheus.bindAddress",
                     )?,
                 };
+
+
+                let options = options.build().expect("Failed to build prometheus exporter options");
+                let prom_info = start_prometheus_metric_exporter(options).expect("Failed creating prometheus exporter");
+                telemetry_opts.metrics(prom_info.meter as Arc<dyn CoreMeter>);
+
             } else if let Some(ref otel) = js_optional_getter!(cx, metrics, "otel", JsObject) {
+                let mut options = OtelCollectorOptionsBuilder::default();
+
                 let url = js_value_getter!(cx, otel, "url", JsString);
-                let url = match Url::parse(&url) {
-                    Ok(url) => url,
-                    Err(_) => cx.throw_type_error("Invalid telemetryOptions.metrics.otel.url")?,
+                match Url::parse(&url) {
+                    Ok(url) => options.url(url),
+                    Err(_) => return cx.throw_type_error("Invalid telemetryOptions.metrics.otel.url"),
+                };
+
+                if let Some(ref headers) = js_optional_getter!(cx, otel, "headers", JsObject) {
+                    options.headers(headers.as_hash_map_of_string_to_string(cx)?);
                 };
-                let headers =
-                    if let Some(ref headers) = js_optional_getter!(cx, otel, "headers", JsObject) {
-                        headers.as_hash_map_of_string_to_string(cx)?
-                    } else {
-                        Default::default()
+
+                if let Some(metric_periodicity) = js_optional_value_getter!(cx, otel, "metricsExportInterval", JsNumber).map(|f| f as u64) {
+                    options.metric_periodicity(Duration::from_millis(metric_periodicity));
+                }
+
+                // FIXME: Move temporality to the otel object
```

Comment:
> Move temporality to the otel object

Don't block on this. It will be fixed in upcoming metrics PR. I want to avoid changing lang side API (and therefore deal with deprecation) just for a single release, then change it again on the next release.